### DISCUSSION TITLE
Moving instance to another server

### DISCUSSION THREAD

**Discussion by fweng322** [mastodon/mastodon#25790] [Q&A] [answered]
Hi,

I'm planning to move my Mastodon instance to another server which has more resources.  Before actually moving I tried to do some experiments.  Let's say to move from server A  to server B.

I installed a new instance on server B first in order to make sure the new server meets the dependencies Mastodon needs.  It is installed under /home/mastodon on server B.

Because I was just doing experiments, so in order not to mess with the original instance (say hostname a.mastodon) I used a new hostname (b.mastodon) when installing.  _(Yes, I know this part may be where the problem is - please read on and I'll ask my question below.)_

Then I used `rsync` to sync the data under /home/mastodon on server A to server B under /home/mastodon.old.  I use `pgdump` to dump the database from server A and transfer the sql file to server B.

Then, on server B, I renamed /home/mastodon to /home/mastodon.new, then renamed /home/mastodon.old to /home/mastodon, making sure the user/group name and permission was correct.  Then I restored the database from server A into server B.

I then edited /home/mastodon/live/.env.production, change the LOCAL_DOMAIN and WEB_DOMAIN from a.mastodon to b.mastodon.

Everything was done, I restarted mastodon services (sidekiq, streaming, web).  I was able to see the site and login into it using my own account, but here is the problem:

*When I logged in, my home timeline was empty!*

The local timeline and global timeline were fine.  The data were all there.  Only my home timeline was gone.

Now, my question is, was this caused by the hostname change?  Or was there any other possibilities which caused my home timeline gone?

I know that when installing the instance it warned us that hostname could not be changed.  I read [this article](https://www.reddit.com/r/Mastodon/comments/ze89cq/anyone_ever_successfully_changed_the_hostname_of/) .  There was a few explanations in it.  So is it the real cause of my timeline gone? 

I mean, if I really couldn't change the hostname, I would need to risk directly changing my DNS settings of the IP of a.mastodon and I was not sure if my migration was successful or not.  (I read [this migration guide](https://docs.joinmastodon.org/admin/migrating/) as well.)  I was just trying to do some experiments without messing up the original site.  So it seems not possible to do such experiments?

Thanks for answering and discussing.

**Comment by justdaj** [comment, marked as answer]
You usually need to rebuild the home timelines

`./bin/tootctl feeds build`
